src/Microsoft.Health.Fhir.Anonymizer.Shared.Core/PartitionedExecution/FhirPartitionedExecutor.cs

```diff
@@ -110,28 +112,40 @@ private async Task<IEnumerable<TResult>> AnonymizeAsync(List<TSource> batchData,
                         throw new OperationCanceledException();
                     }
 
-                    try
+                    TResult anonymizedResult = await AnonymizerFunctionAsync(content);
+                    if (IsEmptyElementResult(anonymizedResult))
                     {
-                        TResult anonymizedResult = await AnonymizerFunctionAsync(content);
-                        result.Add(anonymizedResult);
-                        batchAnonymizeProgressDetail.ProcessCompleted++;
+                        batchAnonymizeProgressDetail.ProcessSkipped++;
                     }
-                    catch (Exception)
+                    else
                     {
-                        if (breakOnAnonymizationException)
-                        {
-                            throw;
-                        }
-
-                        batchAnonymizeProgressDetail.ProcessFailed++;
-                    }                    
+                        batchAnonymizeProgressDetail.ProcessCompleted++;
+                    }
+                    result.Add(anonymizedResult);
                 }
 
                 progress?.Report(batchAnonymizeProgressDetail);
                 return result;
             }).ConfigureAwait(false);
         }
 
+        private bool IsEmptyElementResult(TResult result)
```

Comment:
combine it with IsEmptyElement(ITypedElement element)

Reply:
Fixed it.